File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expression/Id.cs
using AST.Nodes.Interfaces;
using AST.Representation;

namespace AST.Nodes.Expression
{
    public class Id : ASTNode, IExpression, IHasType
    {
        public string Name { get; private set; }
        private Types.Type type = new Types.UndefinedType();
        
        public Id(string name, PositionInText position)
            : base(position)
        {
            this.Name = name;
        }

        public T Accept<T>(ASTVisitors.IVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public Types.Type RetrieveType()
        {
            return this.type;
        }

        public void SetType(Types.Type type)
        {
            this.type = type;
        }

        public override bool Equals(object obj)
        {
            return (this.Name.Equals(((Id)obj).Name));
        }

        public override int GetHashCode()
        {
            return this.Name.GetHashCode();
        }
    }
}

File: Amir/qlProject/src/qlProject/ast/expression/Id.java
package qlProject.ast.expression;

public class Id implements IExpression{

	private final String idString;
	
	public Id (String idString){
		this.idString = idString;
	}
	
	public String getIdString (){
		return this.idString;
	}

	@Override
	public Object accept(IExpressionVisitor visitor) {
		return visitor.visit(this);
	}
}
